src/python/pants/backend/url_handlers/s3/integration_test.py

```diff
@@ -44,11 +45,15 @@ def rule_runner() -> RuleRunner:
     )
 
 
+class NoCredentialsError(Exception):
+    pass
+
+
 @pytest.fixture
 def monkeypatch_botocore(monkeypatch):
-    def do_patching(expected_url):
+    def do_patching(expected_auth_url):
         botocore = SimpleNamespace()
-        botocore.exceptions = SimpleNamespace(NoCredentialsError=Exception)
```

Comment:
Use of Exception here was causing some tests to pass (through this catch https://github.com/pantsbuild/pants/blob/a61754651dd13bd7804a698d6bc3e4531f5467fa/src/python/pants/backend/url_handlers/s3/register.py#L134) that should have been failing. I've updated the cases.